src/cpp/include/openvino/genai/generation_config.hpp

```diff
@@ -91,10 +90,125 @@ class OPENVINO_GENAI_EXPORTS StructuredOutputConfig {
     */
     StructuredOutputConfig(const ov::AnyMap& properties);
     StructuredOutputConfig() = default;
+
+    // base grammar types for compound grammar construction
+    struct Regex {
```

Comment:
There are two types to represent regex: `Regex` here and `std::string` in `StructuredOutputConfig`. Should `StructuredOutputConfig` enable both: `std::variant<std::string, Regex, std::monostate>> regex;`

Reply:
`std::variant<std::string, Regex, std::monostate>> regex` would still leave inconsistency: `StructuredOutputConfig` schema is specified by var name and while `CompoundGrammar` is specified by type

Reply:
I don't think we need to mix two modes here. Types are for compound grammar, and strings are for separate fields (json schema, ebnf grammar and regex all have string representation).